src/server/hook_func.c

```diff
@@ -6751,6 +6944,486 @@ get_hook_rescdef_checksum(void)
+
+			/* important here to look for the leftmost '[' (using strchr)
+			 * and the rightmost ']' (using strrchr)
+			 * as we can have:
+			 *	pbs.event().vnode_list["altix[5]"].<attr>=<val>
+			 * 	and "altix[5]" is a valid vnode id.
+			 */
+			if (((pc1=strchr(obj_name, '[')) != NULL) &&
+				((pc2=strrchr(obj_name, ']')) != NULL) &&
+				(pc2 > pc1)) {
+				pc1++; /*  pc1=<vname>] */
+				*pc2 = '\0'; /* pc1=<vname>  */
+				pc2++;
+
+				/* now let's if there's anything quoted inside */
+				pc3 = strchr(pc1, '"');
+				if (pc3 != NULL)
+					pc4 = strchr(pc3+1, '"');
+				else
+					pc4 = NULL;
+
+				if (pc3 && pc4 && (pc4 > pc3)) {
+					pc3++;
+					*pc4 = '\0';
+					vname_str = pc3;
+				} else {
+					vname_str = pc1;
+				}
+			} else {
+				snprintf(log_buffer, sizeof(log_buffer),
+					"object '%s' does not have a vnode name!",
+					obj_name);
+				log_err(-1, __func__, log_buffer);
+				/* process a new line */
+				line_data[0] = '\0';
+				continue;
+			}
+
+			/* server periodic hook vnode objects */
+			valln = (int)strlen(data_value) + 1;
+			plist = attrlist_create(name_str, resc_str, valln);
+			if (plist == NULL) {
+				(void)sprintf(log_buffer,"failed to add svrattrl list %s.%s.%s:%s",
+						vname_str,name_str,resc_str,data_value);
+				log_event(PBSEVENT_SYSTEM | PBSEVENT_FORCE, PBS_EVENTCLASS_SERVER,
+					LOG_NOTICE, msg_daemonname, log_buffer);
+			} else {
+				strcpy(plist->al_value, data_value);
+				(plist->al_link).ll_next->ll_struct = NULL;
+				/* there are vnode hook updates */
+				/* Push hook changes to server */
+				pnode = find_nodebyname(vname_str);
+				if (pnode == NULL) {
+					log_event(PBSEVENT_DEBUG3, PBS_EVENTCLASS_HOOK,
+						LOG_INFO, phook->hook_name, "node_name not found");
+				} else if ((pnode->nd_state & INUSE_DELETED) == 0) {
+					save_characteristic(pnode);
+
+					rc = mgr_set_attr(pnode->nd_attr, node_attr_def, ND_ATR_LAST,
+								  plist, ATR_DFLAG_WRACC, &bad, (void *)pnode, ATR_ACTION_ALTER);
+					if (rc != 0) {
+						pbse_err = pbse_to_txt(rc);
+						snprintf(raw_err, sizeof(raw_err), "%d", rc);
+						sprintf(log_buffer, "vnode %s: failed to set %s to %s: %s",
+								pnode->nd_name, plist->al_name, plist->al_value ? plist->al_value : "",
+								pbse_err ? pbse_err : raw_err);
+						log_err(PBSE_SYSTEM, __func__, log_buffer);
+					} else {
+						if ((pnode->nd_nsnfree == 0) && (pnode->nd_state == 0))
+							set_vnode_state(pnode, INUSE_JOB, Nd_State_Or);
+
+						(void)chk_characteristic(pnode, &ndtype_flag);
+						mgr_log_attr(msg_man_set, plist,
+								 PBS_EVENTCLASS_NODE, pnode->nd_name, NULL);
+					}
+				}
+				if (ndtype_flag & WRITE_NEW_NODESFILE) {
+					/*create/delete/prop/ntype change*/
+					(void)save_nodes_db(0, NULL);
+				} else if (ndtype_flag & WRITENODE_STATE) {
+					write_node_state();
+				}
+			}
+		} else if ((strcmp(obj_name, SERVER_OBJECT) == 0) &&
+			(strcmp(name_str, PY_SCHEDULER_RESTART_CYCLE_METHOD) == 0)) {
```

Comment:
why are you checking for restart cycle method. Rather, don't add string comparisons that we do not want to implement right now.

Reply:
Agreed. Removed all TODO sections i had put.